src/job/MomoJobBuilder.ts

```diff
@@ -0,0 +1,53 @@
+import { Handler, MomoJob } from './MomoJob';
+
+export class MomoJobBuilder {
+  private momoJob: Partial<MomoJob> = {};
+
+  withName(name: string): this & Required<Pick<MomoJob, 'name'>> {
```

Comment:
aha, this is actually a mistake - the return type was supposed to be adjusted. I'll fix that now

Reply:
Do you have the newest version? My version reads
`  withName(name: string): this & Required<Pick<MomoJob, 'name'>> {
    this.momoJob.name = name;
    return this as this & Required<Pick<MomoJob, 'name'>>;
  }`

Reply:
Implementing the type-safety in the way I've been trying to do an inherent conflict with keeping the class fields private, since these need to be visible from outside the class in order to be checked. Since keeping the attributes of a builder class private is an important part of the pattern, I've decided to drop the type-safety for now.